crates/uv-distribution-types/src/prioritized_distribution.rs

```diff
@@ -555,6 +556,60 @@ impl PrioritizedDist {
         }
         candidates
     }
+
+    /// Return the "closest" tag for the distribution that is compatible with the given tags.
+    ///
+    /// The tag must match at least two of the three components (Python, ABI, platform) in order to
+    /// be considered compatible. If multiple tags are compatible, the tag with the highest priority
+    /// is returned. If multiple tags have the same priority, the tag with the highest version is
+    /// returned.
+    ///
+    /// For example, if `cp313-cp313-macosx_10_9_x86_64` is compatible (but not available), then
+    /// `cp313-cp313-manylinux2010_x86_64` could be returned.
+    pub fn closest_tag(&self, tags: &Tags) -> Option<(LanguageTag, AbiTag, String)> {
+        let mut closest: Option<(TagPriority, LanguageTag, AbiTag, &str)> = None;
+        for (py, abi, platform, priority) in tags.iter() {
+            // If this tag is lower-priority than the current closest tag, skip it.
+            if let Some((existing_priority, ..)) = closest {
+                if priority < existing_priority {
+                    continue;
+                }
+            }
+            let Ok(py) = LanguageTag::from_str(py) else {
+                continue;
+            };
+            let Ok(abi) = AbiTag::from_str(abi) else {
+                continue;
+            };
+            for (wheel, _) in &self.0.wheels {
```

Comment:
This is really, really inefficient... It's O(M*N) where M is the number of compatible tags and N is the number of wheels. But like... it will only run once, in the error reporting, so I think it's kind of fine?